include/deal.II/base/tensor.h

```diff
@@ -1213,21 +1164,6 @@ Tensor<rank_,dim,Number>::serialize(Archive &ar, const unsigned int)
 
 /* ----------------- Non-member functions operating on tensors. ------------ */
 
-
-#ifndef DEAL_II_WITH_CXX11
-template <typename T, typename U, int rank, int dim>
-struct ProductType<T,Tensor<rank,dim,U> >
-{
```

Comment:
I think these are still relevant as long as `T` is a a scalar. Can we model this somehow?